### ISSUE TITLE
New Post Page Followups

### ISSUE THREAD

**Issue by micahmo** [thunder-app/thunder#1413]
### Bug Description

This issue is for tracking additional work that needs to be done to the new post/comment page as a followup to #1363 in order to bring it in parity with the old one. The new one is still behind an experimental flag, so these things aren't super urgent, but should be done before releasing it for general availability.

#### Known Issues (these are things intentionally deferred from the initial implementation)
* [x] Bring back FAB
* [x] Bring back comment navigation
	* [x] Allow comment navigation to navigate between top level comments, and also child comments
	* [x] Ensure that the first navigation goes to the top of the first comment (see https://github.com/thunder-app/thunder/pull/1496#issuecomment-2263800481)
* [x] Bring back comment search - with the ability to scroll to specific child comments

#### Additional Issues (these have been found while daily driving the new page)
- [x] Collapsing comment threads does not show any animation (for the child comments being hidden)
<details>

https://github.com/thunder-app/thunder/assets/7417301/7f0b3b76-9df4-4c41-8fb7-a028f3cd69e8

</details>

- [x] Collapsing a comment thread with only deferred child comments does not hide the placeholder (this is especially confusing because if you try to load the deferred comments after hiding them, they'll disappear!)
<details>

https://github.com/thunder-app/thunder/assets/7417301/7f0b3b76-9df4-4c41-8fb7-a028f3cd69e8

</details>

- [x] Child comment threads are indented too much
<details>

![image](https://github.com/thunder-app/thunder/assets/7417301/3be61961-b3ff-4951-99f9-cdcd1e838e02)

</details>

- [x] Using the "navigate to post" feature after creating a post doesn't work

<details>

https://github.com/thunder-app/thunder/assets/7417301/8f2f5304-40b7-43c7-b6cd-c3cf99b55d50

</details>

- [x] Jumping to comment context (e.g., from notifications) shows the old page
- [x] Threads with many comments show a spinner at the bottom (maybe paging isn't working?)

<details>

https://github.com/thunder-app/thunder/assets/7417301/f2fd31b4-9b7b-474d-b854-89a8d0be4c03

</details>

- [x] There is a weird gap in the cross-posts widget

<details>

https://github.com/thunder-app/thunder/assets/7417301/f4ab620f-a847-4448-9382-8ab926c46a98

</details>

- [x] There is no pull-down to refresh (this isn't a big deal, and if the pulling down is either contradicting with another gesture or causing lots of rebuilds, maybe we don't need it)
  - Actually it looks like the refresh button doesn't work either
- [x] The scrolling fixes from [1252](https://github.com/thunder-app/thunder/pull/1252) and [1319](https://github.com/thunder-app/thunder/pull/1319) seem to be missing. It should be possible to scroll the last comment to the top of the screen.
- [x] When you create a comment, it does get displayed immediately, but it's not highligted.

#### On a bright note, I believe the following issues have been resolved (either from #1363 or #1361).
- [x] Swipe anywhere to go back works much more consistently (#198)
- [x] I haven't seen the random stretching issue (#461)

**Comment by hjiangsu** [comment]
> The scrolling fixes from https://github.com/thunder-app/thunder/pull/1252 and https://github.com/thunder-app/thunder/pull/1319 seem to be missing

I was planning to add this back in in conjunction with the comment navigation!

**Comment by micahmo** [comment]
> > The scrolling fixes from #1252 and #1319 seem to be missing
> 
> I was planning to add this back in in conjunction with the comment navigation!

Makes sense! Just as long as you can scroll the last comment to the top even without using the nav buttons. 😊😊

**Comment by micahmo** [comment]
Hey @hjiangsu the last unchecked item here is collapse/expand animation, which I see is working now, so I think we can consider this issue complete!
